### DISCUSSION TITLE
Birth location of an object changes with different time ranges

### DISCUSSION THREAD

**Discussion by graphenoid** [jobovy/galpy#553] [Q&A]
I initialized the orbit of an object with the following code:

```
cl = SkyCoord(ra=259.103*u.deg,
              dec=-39.316*u.deg,
              distance=2.53*u.kpc,
              pm_ra_cosdec=-0.455*u.mas/u.yr,
              pm_dec=-1.439*u.mas/u.yr,
              radial_velocity=-14.423*u.km/u.s, 
              frame='icrs')
o = Orbit(cl)
# ob = Orbit(cl)
logAge = 7.0
clage = 10**(logage-9)#*u.Gyr

ts = np.linspace(-200.0, 200.0, 10001)#*u.Gyr
o.integrate(ts, MWPotential2014)
```

and I calculate the birth location in the (X, Y, Z) space as follows:

```
o.x(-clage), o.y(-clage), o.z(-clage)
```
which gives the following plot:
![download](https://user-images.githubusercontent.com/39805300/219973456-47eccb32-773d-443b-a504-40735a494423.png)

when I change the integration time as follows:
```
ts = np.linspace(-100.0, 100.0, 10001)#*u.Gyr
```
the plots look like this:
![download](https://user-images.githubusercontent.com/39805300/219973600-1284bf33-f696-4a03-a433-a1207d70d61f.png)

the green stars indicates the birth location, whereas the red dot is the current location of the object.
When I multiply the integration time by u.Gyr, it doesn't change the result. The birth location changes everytime.
The version of galpy is 1.8.0 

How can I find the birth location of an object?

Best,
Sivan.

**Comment by jobovy** [comment]
Hi!

The first entry in the time array is the time of the initial condition (that is, the initial condition isn't assumed to be at `t=0`, which I think you may have thought). So because you change the initial time, your `clage` will be a different time in the past. If you keep the initial times the same, e.g., in your second run use
```
ts = np.linspace(-200.0, 100.0, 10001)#*u.Gyr
```
then you'll see that you get a consistent result.

A few more comments:
* You do have to add `*u.Gyr` if you want your times to be interpreted as being in Gyr. Without this, the times are assumed to be in internal units. Given how you define `clage`, I think this is supposed to be in Gyr.
* I think you want to integrate backwards in time, but you are integrating forwards in time. For your purposes, the best thing to do would be to use
  ```
  ts = np.linspace(0,-clage, 10001)
  ```
  So you go from the present (defined to be `t=0` here) to the time you want to integrate backwards for.

I hope this helps! Please note that I have converted this Issue to a Discussion, because this isn't a bug in the code. Please label the answer as the correct one if this resolves your question.

**Comment by graphenoid** [comment]
Hi,

Thank you for the quick response! But what I don't understand is why setting different iteration time intervals gives different results for the same time in the past (in this case `clage`) even though `clage` is in the iteration range.

I converted the time to Gyr:
 ```
cl = SkyCoord(ra=259.047*u.deg,
              dec=-39.429*u.deg,
              distance=2.41*u.kpc,
              pm_ra_cosdec=-0.78*u.mas/u.yr,
              pm_dec=-2.47*u.mas/u.yr,
              radial_velocity=-38.04*u.km/u.s, 
              frame='icrs')
clage = 0.16*u.Gyr
o = Orbit(cl)
ts = np.linspace(-10.0, 10.0, 1001)*u.Gyr
o.integrate(ts, MWPotential2014)
```
As you see, `clage` is in the range of `ts` and when I overplot `o.x(-clage), o.y(-clage), o.z(-clage)`, I get the following plot.
![download](https://user-images.githubusercontent.com/39805300/220173965-4cf2f024-9eab-426d-846d-c4804a7fbbd6.png)

when I set `ts = np.linspace(-5.0, 5.0, 1001)*u.Gyr`, still `clage` is in the range, I get this:
![download](https://user-images.githubusercontent.com/39805300/220174212-07c65615-07bc-4e5b-ab92-5c53840789a9.png)

I also tried to integrate the time in the opposite direction such as `ts = np.linspace(5.0, -5.0, 1001)*u.Gyr` and `ts = np.linspace(10.0, -10.0, 1001)*u.Gyr`, still get the different results. `o.x(-clage), o.y(-clage), o.z(-clage)` always give different positions. 

`ts = np.linspace(12.0, -clage/u.Gyr, 1001)*u.Gyr` and `ts = np.linspace(10.0, -clage/u.Gyr, 1001)*u.Gyr` also give different results.

if I use `ts = np.linspace(0.0, -clage/u.Gyr, 1001)*u.Gyr`, I sometimes won't get a closed orbit due to the small `clage` values.

What I actually want to get is an orbit being both backward and forward in time, and mark where the object was born, and the current locations on the plot.

all the best,
Sivan.
